cpp/src/parquet/metadata.cc

```diff
@@ -394,6 +394,43 @@ class FileMetaData::FileMetaDataImpl {
     }
   }
 
+  format::RowGroup& get_row_group(int i) {
+    if (!(i < num_row_groups())) {
+      std::stringstream ss;
+      ss << "The file only has " << num_row_groups()
+         << " row groups, requested metadata for row group: " << i;
+      throw ParquetException(ss.str());
+    }
+    return metadata_->row_groups[i];
+  }
+
+  void AppendRowGroups(const std::unique_ptr<FileMetaDataImpl>& other) {
+    format::RowGroup other_rg;
+    for (int i=0; i<other->num_row_groups(); i++) {
+      other_rg = other->get_row_group(i);
+      metadata_->row_groups.push_back(other_rg);
+      metadata_->num_rows += other_rg.num_rows;
+    }
+  }
+
+  void WriteMetaDataFile(OutputStream* sink) {
```

Comment:
there is already a `WriteFileMetaData` in file-writer.cc. That function can't be reused? (does it need to be duplicated here?)